velox/exec/Driver.cpp

```diff
@@ -591,13 +565,9 @@ StopReason Driver::runInternal(
           // for the current running arbitration, it is more efficient
           // system-wide to let driver go off thread for the other queries which
           // have free memory capacity to run during the time.
-          blockedOperatorId_ = curOperatorId_ + 1;
-          VELOX_CHECK(future.valid());
           blockingReason_ = BlockingReason::kWaitForArbitration;
-          blockingState = std::make_shared<BlockingState>(
-              self, std::move(future), op, blockingReason_);
           guard.notThrown();
-          return StopReason::kBlock;
+          return blockDriver(self, op, std::move(future), blockingState);
```

Comment:
`blockDriver` must be called before `guard.notThrown()`

Reply:
Sure I moved guard.notThrown() into blockDriver to be after creating blockingState: 
```
StopReason Driver::blockDriver(
    const std::shared_ptr<Driver>& self,
    Operator* op,
    ContinueFuture&& future,
    std::shared_ptr<BlockingState>& blockingState,
    CancelGuard& guard) {
  VELOX_CHECK(
      future.valid(),
      "The operator {} is blocked but blocking future is not "
      "set by isBlocked method.",
      op->operatorType());

  blockedOperatorId_ = curOperatorId_ + 1;
  blockingState = std::make_shared<BlockingState>(
      self, std::move(future), op, blockingReason_);
  guard.notThrown();
  return StopReason::kBlock;
}
```